projects/packages/videopress/src/client/block-editor/hooks/use-sync-media/index.ts

```diff
@@ -169,6 +166,26 @@ export function useSyncMedia(
 
 	const isSaving = useSelect( select => select( editorStore ).isSavingPost(), [] );
 	const wasSaving = usePrevious( isSaving );
+
+	// In native, it's not currently possible to access a post's saved state from the editor store.
+	// We therefore need to listen to a native event emitter to know when a post has just been saved.
+	const [ postHasBeenJustSavedNative, setPostHasBeenJustSavedNative ] = useState( false );
+
+	isNative &&
+		useEffect( () => {
+			const handlePostSaveEvent = () => {
```

Comment:
As noted in [the relevant Gutenberg PR](https://github.com/WordPress/gutenberg/pull/49781), we could potentially connect this to [Gutenberg's isSavingPost selector](https://github.com/WordPress/gutenberg/blob/c475dcb3ffee7627a8e1d2f13f364a7a9b195a61/packages/editor/src/store/selectors.js#L677-L692) in the future. For now, I've proposed this solution in the interest of time and simplicity.